opendp/smartnoise/core/value.py

```diff
@@ -124,20 +124,20 @@ def serialize_indexmap_release_node(release_values):
     )
 
 
-def detect_atomic_type(dtype):
-    if np.issubdtype(dtype, np.integer):
+def detect_atomic_type(array):
```

Comment:
IMO this will be less error-prone over its maintenance lifetime if implemented with either a switch/match with multiple return statements or with a series of if/else blocks with a single return at the function's conclusion.

Reply:
Thanks for the comment. I pushed a commit that changes this.